src/services/cleeng.account.service.ts

```diff
@@ -29,238 +31,272 @@ import type {
 
-  return {
-    errors: [],
-    responseData,
-  };
-};
+    const { responseData, errors } = await this.getCustomer({ customerId: customer.id }, sandbox);
+    this.handleErrors(errors);
 
-export const resetPassword: ResetPassword = async (payload, sandbox) => {
-  return put(sandbox, '/customers/passwords', JSON.stringify(payload));
-};
+    return {
+      errors: [],
+      responseData,
+    };
+  };
 
-export const changePasswordWithResetToken: ChangePassword = async (payload, sandbox) => {
-  return patch(sandbox, '/customers/passwords', JSON.stringify(payload));
-};
+  resetPassword: ResetPassword = async (payload, sandbox) => {
+    return this.cleengService.put(sandbox, '/customers/passwords', JSON.stringify(payload));
+  };
 
-export const changePasswordWithOldPassword: ChangePasswordWithOldPassword = async () => {
-  return {
-    errors: [],
-    responseData: {},
+  changePasswordWithResetToken: ChangePassword = async (payload, sandbox) => {
+    return this.cleengService.patch(sandbox, '/customers/passwords', JSON.stringify(payload));
   };
-};
 
-export const updateCustomer: UpdateCustomer = async (payload, sandbox) => {
-  const { id, metadata, fullName, ...rest } = payload;
-  const params: UpdateCustomerPayload = {
-    id,
-    ...rest,
+  changePasswordWithOldPassword: ChangePasswordWithOldPassword = async () => {
+    return {
+      errors: [],
+      responseData: {},
+    };
   };
-  // enable keepalive to ensure data is persisted when closing the browser/tab
-  return patch(sandbox, `/customers/${id}`, JSON.stringify(params), { authenticate: true, keepalive: true });
-};
 
-export const getCustomer: GetCustomer = async (payload, sandbox) => {
-  return get(sandbox, `/customers/${payload.customerId}`, { authenticate: true });
-};
+  updateCustomer: UpdateCustomer = async (payload, sandbox) => {
+    const { id, metadata, fullName, ...rest } = payload;
+    const params: UpdateCustomerPayload = {
+      id,
+      ...rest,
+    };
+    // enable keepalive to ensure data is persisted when closing the browser/tab
+    return this.cleengService.patch(sandbox, `/customers/${id}`, JSON.stringify(params), { authenticate: true, keepalive: true });
+  };
 
-export const getLocales: GetLocales = async (sandbox) => {
-  return get(sandbox, `/locales${getOverrideIP() ? '?customerIP=' + getOverrideIP() : ''}`);
-};
+  updatePersonalShelves: UpdatePersonalShelves = async (payload, sandbox) => {
+    return await this.updateCustomer(payload, sandbox);
+  };
 
-const handleErrors = (errors: ApiResponse['errors']) => {
-  if (errors.length > 0) {
-    throw new Error(errors[0]);
-  }
-};
+  subscribeToNotifications: NotificationsData = async () => {
+    return false;
+  };
 
-export const updatePersonalShelves: UpdatePersonalShelves = async (payload, sandbox) => {
-  return await updateCustomer(payload, sandbox);
-};
+  getSocialUrls: SocialURLSData = async () => {
+    throw new Error('Method is not supported');
+  };
```

Comment:
Throwing runtime errors or returning dummy data is not a great way to enforce what is allowed by each service, because it masks mistakes at build time which will only surface at runtime.

```suggestion
  getSocialUrls: undefined
```